packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -97,18 +97,22 @@ Future<GradleProject> _readGradleProject() async {
   final Status status = logger.startProgress('Resolving dependencies...', expectSlowOperation: true);
   GradleProject project;
   try {
-    final RunResult runResult = await runCheckedAsync(
+    final RunResult propertiesRunResult = await runCheckedAsync(
       <String>[gradle, 'app:properties'],
       workingDirectory: flutterProject.android.hostAppGradleRoot.path,
       environment: _gradleEnv,
     );
-    final String properties = runResult.stdout.trim();
-    project = GradleProject.fromAppProperties(properties);
+    final RunResult tasksRunResult = await runCheckedAsync(
+      <String>[gradle, 'app:tasks', '--all'],
+      workingDirectory: flutterProject.android.hostAppGradleRoot.path,
+      environment: _gradleEnv,
+    );
```

Comment:
I'm not familiar enough with Gradle's history, but I assume `app:tasks --all` support goes back to whatever older versions we've previously supported? Otherwise we could add a doctor check.

Reply:
It looks like `tasks --all` has been supported for several years.

I tried implementing a version of this that invokes `app:properties` and `app:tasks --all` in one invocation to avoid the cost of launching two Gradle processes.  I did notice that the output format used when running multiple targets in one command is inconsistent across Gradle versions.